cmake/multi_image.cmake

```diff
@@ -29,6 +29,9 @@ if(IMAGE_NAME)
   share("list(APPEND ${IMAGE_NAME}BUILD_BYPRODUCTS ${PROJECT_BINARY_DIR}/${KERNEL_ELF_NAME})")
   # Share the signing key file so that the parent image can use it to
   # generate signed update candidates.
+  if(CONFIG_UPDATEABLE_IMAGE_NUMBER)
+    share("set(${IMAGE_NAME}UPDATEABLE_IMAGE_NUMBER ${CONFIG_UPDATEABLE_IMAGE_NUMBER})")
```

Comment:
@tejlmand  since https://github.com/zephyrproject-rtos/zephyr/blob/7c2dfc1558e598e07c015738e984f5c50c3e6c82/subsys/dfu/boot/mcuboot.c#L19 has started to use the bootutil_public API we need a way of sharing configuration between MCUBoot and Application since changes in MCUBoot config are not propagated to the application sourcing of `mcuboot_config.h` here https://github.com/mcu-tools/mcuboot/blob/78e4441bd373b2ea8a7c36787ee657953c65481f/boot/bootutil/include/bootutil/bootutil_public.h#L44

As you can see in 
https://github.com/mcu-tools/mcuboot/blob/main/boot/zephyr/include/mcuboot_config/mcuboot_config.h
there area lot of configurations sourced from the Kconfig which are not reflected in the application sourcing of `mcuboot_config.h` since they will not be defined as they do not exists.

I don't know if you have any good idea how to do this or if we should just try to keep adding the configurations we need to be set. 

For multi-image applications that would making sure `CONFIG_UPDATEABLE_IMAGE_NUMBER > 1` but also we need the coming  hooks API(https://github.com/mcu-tools/mcuboot/pull/1058). Where we need to make sure `MCUBOOT_IMAGE_ACCESS_HOOKS` is set by propagating the configuration.

1. Solution is to mirror the Kconfigs and set them also in application image.